src/NServiceBus.RavenDB/SessionManagement/RavenSessionExtension.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Raven.Client;
+
+namespace NServiceBus.RavenDB.Persistence
+{
+    using System;
+
+    /// <summary>
+    /// Extensions, to the message handler context, to manage RavenDB session.
+    /// </summary>
+    public static class RavenSessionExtension
+    {
+        /// <summary>
+        /// Gets the current RavenDB session.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <returns></returns>
+        public static IAsyncDocumentSession GetRavenSession(this IMessageHandlerContext context)
+        {
+            Func<IAsyncDocumentSession> sessionFunction;
+            context.Extensions.TryGet(out sessionFunction);
+            if (sessionFunction != null)
+            {
+                return sessionFunction();
+            }
+            IAsyncDocumentSession session;
+            context.Extensions.TryGet(out session);
+            if (session == null)
+            {
+                throw new Exception("Could not retrieve a RavenDB session. Please ensure that you are using a saga or have enabled outbox.");
```

Comment:
Not good. Doesn't make me understand why that would fix it.

GetRavenSession() allows retrieval of the shared Raven IAsyncDocumentSession being used by NServiceBus so that additional Raven operations can be completed in the same transactional context as NServiceBus Sagas and Outbox features. Because the Saga and Outbox features are not currently in use, it was not possible to retrieve a RavenDB session.

How's that?